frontend/src/pages/SignupPage.js

```diff
@@ -0,0 +1,113 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { useState } from "react";
+import axios from "axios";
+import { URL_USER_SVC } from "../configs";
+import { STATUS_CODE_CONFLICT, STATUS_CODE_CREATED } from "../constants";
+import { Link } from "react-router-dom";
+
+function SignupPage() {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [dialogTitle, setDialogTitle] = useState("");
```

Comment:
Similar to the username and password issue, it would be better to put the dialog title and dialog message together as an object.